elements/pfe-navigation/src/pfe-navigation-item.js

```diff
@@ -0,0 +1,429 @@
+        if (i in this){
+          kValue = t[i]; // in case t is changed in callback
+          if (func(t[i], i, t)){
+            res[c++] = kValue;
+          }
+        }
+      }
+    }
+    else{
+      while (++i !== len){
+        // checks to see if the key was set
+        if (i in this){
+          kValue = t[i];
+          if (func.call(thisArg, t[i], i, t)){
+            res[c++] = kValue;
+          }
+        }
+      }
+    }
+   
+    res.length = c; // shrink down array to proper size
+    return res;
+  };
+}
+
+class PfeNavigationItem extends PFElement {
+    static get tag() {
+      return "pfe-navigation-item";
+    }
+  
+    get templateUrl() {
+      return "pfe-navigation-item.html";
+    }
+  
+    get styleUrl() {
+      return "pfe-navigation-item.scss";
+    }
+  
+    get schemaUrl() {
+      return "pfe-navigation-item.json";
+    }
+  
+    static get PfeType() {
+      return PFElement.PfeTypes.Container;
+    }
+  
+    static get observedAttributes() {
+      return ["pfe-icon"];
+    }
+
+    get hasIcon() {
+      return this.hasAttribute("pfe-icon");
+    }
+
+    get iconName() {
+      return this.getAttribute("pfe-icon");
+    }
+  
+    get nested() {
+      return this.hasAttribute("is_nested");
+    }
+  
+    set nested(isNested) {
+      isNested = Boolean(isNested);
+  
+      if (isNested) {
+        this.setAttribute("is_nested", "");
+      } else {
+        this.removeAttribute("is_nested");
+      }
+    }
+  
+    get expanded() {
+      return this.classList.contains("expanded");
+    }
+  
+    set expanded(isExpanded) {
+      isExpanded = Boolean(isExpanded);
+  
+      if (isExpanded) {
+        this.classList.add("expanded");
+  
+        if (this.iconName === "web-mobile-menu") {
+          if (this._icon) this._icon.setAttribute("icon", "web-plus");
+        }
+
+        if (this._trigger) {
+          this._trigger.setAttribute("aria-expanded", true);
+        }
+  
+        if (this.tray) {
+          this.tray.removeAttribute("hidden");
+        }
+  
+        if (this._tray) {
+          this._tray.setAttribute("aria-expanded", true);
+        }
+      } else {
+        this.classList.remove("expanded");
+  
+        if (this.iconName === "web-mobile-menu") {
+          if (this._icon) this._icon.setAttribute("icon", "web-mobile-menu");
+        }
+  
+        if (this._trigger) {
+          this._trigger.setAttribute("aria-expanded", false);
+        }
+  
+        if (this.tray) {
+          this.tray.setAttribute("hidden", "");
+        }
+  
+        if (this._tray) {
+          this._tray.setAttribute("aria-expanded", false);
+        }
+      }
+    }
+  
+    get visible() {
+      return !this.hasAttribute("hidden");
+    }
+  
+    set visible(isVisible) {
+      isVisible = Boolean(isVisible);
+  
+      if (isVisible) {
+        this.removeAttribute("hidden");
+      } else {
+        this.setAttribute("hidden", "");
+      }
+    }
+  
+    open(event) {
```

Comment:
This is super nit picky but can we move these public API methods after the `disconnectedCallback`? It just makes things consistent with the other components.

Reply:
We can but I think it makes sense to keep public-facing functions at the top because those are the ones developers will be looking at most frequently during debugging.